inference-engine/samples/common/utils/src/args_helper.cpp

```diff
@@ -326,3 +356,94 @@ void printInputAndOutputsInfo(const InferenceEngine::CNNNetwork& network) {
                   << layer.second->getLayout() << std::endl;
     }
 }
+
+void configurePrePostProcessing(std::shared_ptr<ov::Function>& function,
+                                const std::string& ip,
+                                const std::string& op,
+                                const std::string& iop,
+                                const std::string& il,
+                                const std::string& ol,
+                                const std::string& iol) {
+    auto preprocessor = ov::preprocess::PrePostProcessor();
+    if (!ip.empty()) {
+        auto type = getType(ip);
+        for (const auto& input : function->inputs()) {
+            preprocessor.input(ov::preprocess::InputInfo(input.get_any_name())
+                                   .tensor(ov::preprocess::InputTensorInfo().set_element_type(type)));
+        }
+    }
+
+    if (!op.empty()) {
+        auto type = getType(op);
+        for (const auto& output : function->outputs()) {
+            preprocessor.output(ov::preprocess::OutputInfo(output.get_any_name())
+                                    .tensor(ov::preprocess::OutputTensorInfo().set_element_type(type)));
+        }
+    }
+
+    if (!iop.empty()) {
+        const auto user_precisions_map = parseArgMap(iop);
+        for (auto&& item : user_precisions_map) {
+            const auto& tensor_name = item.first;
+            const auto type = getType(item.second);
+
+            bool isInput = false;
+            for (const auto& input : function->inputs()) {
+                if (input.get_names().count(tensor_name)) {
+                    preprocessor.input(ov::preprocess::InputInfo(input.get_any_name())
+                                           .tensor(ov::preprocess::InputTensorInfo().set_element_type(type)));
+                    isInput = true;
+                    break;
+                }
+            }
+            if (!isInput) {
+                for (const auto& output : function->outputs()) {
+                    if (output.get_names().count(tensor_name)) {
+                        preprocessor.output(ov::preprocess::OutputInfo(output.get_any_name())
+                                                .tensor(ov::preprocess::OutputTensorInfo().set_element_type(type)));
+                    }
+                }
+            }
+        }
+    }
+    if (!il.empty()) {
+        for (auto&& input : function->inputs()) {
+            preprocessor.input(ov::preprocess::InputInfo(input.get_any_name())
+                                   .tensor(ov::preprocess::InputTensorInfo().set_layout(ov::Layout(il))));
```

Comment:
Shouldn't we setup network layout as well? Or it would be assumed as default NCHW?

Reply:
It can be different and user should define layout in the IR, or we need to introduce additional command line arguments.

Reply:
> or we need to introduce additional command line arguments.

I think we need it, otherwise only tensor layout does not lead to any preprocessing insertion